sourced/ml/algorithms/uast_struct_to_bag.py

```diff
@@ -1,24 +1,32 @@
 import random
 from collections import defaultdict, deque
 
-from sourced.ml.algorithms.uast_ids_to_bag import FakeVocabulary
+from sourced.ml.algorithms.uast_ids_to_bag import FakeVocabulary, Uast2BagBase
 
 
-class UastStructure2BagBase:
-    def uast_to_bag(self, uast):
-        raise NotImplemented
+class Uast2StructBagBase(Uast2BagBase):
+    def __init__(self, stride, seq_len, node2index=None):
+        self._node2index = node2index if node2index is not None else FakeVocabulary()
+        self._stride = stride
+        if not isinstance(seq_len, (int, tuple, list)):
+            raise TypeError("Unexpected type of seq_len: %s" % type(seq_len))
+        self._seq_lens = [seq_len] if isinstance(seq_len, int) else seq_len
+
+    @property
+    def node2index(self):
+        return self._node2index
 
 
-class UastSeq2Bag(UastStructure2BagBase):
+class UastSeq2Bag(Uast2StructBagBase):
     """
     DFS traversal + preserves the order of node children.
     """
-    def __init__(self, type2ind=None, stride=1, seq_len=5):
-        self.type2ind = type2ind if type2ind is not None else FakeVocabulary()
-        self.stride = stride
-        self.seq_len = seq_len
+    def __init__(self, stride=1, seq_len=5, node2index=None):
+        _node2index = Node2InternalType() if node2index is None else node2index
```

Comment:
`Node2InternalType ` definition is below (https://github.com/src-d/ml/pull/145/files#diff-86f78a3692ae357bb5a7628fcf0cbc6bR70). Think that I need to lift it up before this class.